### ISSUE TITLE
Traefik refusing connections while reloading config

### ISSUE THREAD

**Issue by FairyTail2000** [traefik/traefik#9939]
### Welcome!

- [X] Yes, I've searched similar issues on [GitHub](https://github.com/traefik/traefik/issues) and didn't find any.
- [X] Yes, I've searched similar issues on the [Traefik community forum](https://community.traefik.io) and didn't find any.

### What did you do?

While traefik is reloading the docker swarm config, no new clients can access the redis instance hosted behind it. We noticed this extensively since we have hundreds of connections which sometimes reported the error: "Connection refused" which seems to coincide with traefik reloading the swarm config

### What did you see instead?

Connection refused messages which resolved themself after the automatic retry

### What version of Traefik are you using?
```
Version:      3.0.0-beta2
Codename:     beaufort
Go version:   go1.19.4
Built:        2022-12-07T16:32:34Z
OS/Arch:      linux/amd64
```
### What is your environment & configuration?

```yaml
entryPoints:
  web:
    address: :80
    http:
      redirections:
        entryPoint:
          to: websecure
          permanent: true

  websecure:
    address: :443
  postgres:
    address: :5432
  redis:
    address: :6379
providers:
  file:
    filename: /etc/traefik/traefik.yaml 
    watch: true
  docker:
    exposedByDefault: false
    endpoint: unix:///var/run/docker.sock
    #endpoint: "tcp://127.0.0.1:2377"
    swarmMode: true
    network: traefik_traefik-network
```

```yaml
    deploy:
        labels:
                - traefik.tcp.routers.rediss.rule=HostSNI(`*`)
                - traefik.tcp.services.rediss.loadbalancer.server.port=6379
                - traefik.tcp.routers.rediss.service=rediss
                - traefik.tcp.routers.rediss.entrypoints=redis
                - traefik.tcp.routers.rediss.tls.passthrough=true
                - traefik.docker.network=traefik_traefik-network
                - traefik.enable=true
```

### If applicable, please paste the log output in DEBUG level
```
2023-06-08T14:01:08Z DBG github.com/traefik/traefik/v2/pkg/server/configurationwatcher.go:217 > Configuration received config=censored providerName=docker
2023-06-08T14:01:08Z DBG github.com/traefik/traefik/v2/pkg/server/configurationwatcher.go:126 > Skipping unchanged configuration providerName=docker
2023-06-08T14:01:10Z DBG github.com/traefik/traefik/v2/pkg/tcp/proxy.go:51 > Handling connection from X.X.X.X:37748 to 10.0.1.21:6379
2023-06-08T14:01:10Z DBG github.com/traefik/traefik/v2/pkg/tcp/proxy.go:51 > Handling connection from X.X.X.X:58092 to 10.0.1.21:6379
2023-06-08T14:01:10Z DBG github.com/traefik/traefik/v2/pkg/tcp/proxy.go:51 > Handling connection from X.X.X.X:44496 to 10.0.1.21:6379
2023-06-08T14:01:10Z DBG github.com/traefik/traefik/v2/pkg/tcp/proxy.go:51 > Handling connection from X.X.X.X:36584 to 10.0.1.21:6379
2023-06-08T14:01:10Z DBG github.com/traefik/traefik/v2/pkg/tcp/proxy.go:51 > Handling connection from X.X.X.X:48028 to 10.0.1.21:6379
2023-06-08T14:01:10Z DBG github.com/traefik/traefik/v2/pkg/tcp/proxy.go:51 > Handling connection from X.X.X.X:41176 to 10.0.1.21:6379
```

**Comment by ldez** [comment]
Hello,

there is no reload with Traefik: the configuration is built before being applied.
Traefik reads the information from the provider to build the configuration but during this build the routing is not changed or impacted.
When the configuration is built, Traefik changes the routing in memory, this takes a few nanoseconds.

Your problem can be related to memory or CPU: during the build of the configuration, the memory and CPU consumption will increase.

I recommend monitoring the memory and CPU consumption.

**Comment by FairyTail2000** [comment]
Hi there,

Thanks for the response.

Upon checking in with lazydocker on the resource usage, it shows that at no point it surpasses 80 MB of RAM. The server I'm running it on has 2 GB of available, non-cache and non committed ram.

So memory shouldn't be an issue. I did however defined a limit in the compose file I'm using to deploy it.

I set the reservation to 80mb and the limit to 150mb. I have now removed and updated the service and will report back if the issue persists
